pkg/plugins/shell/errors.go

```diff
@@ -1,27 +1,13 @@
 package shell
 
-import (
-	"fmt"
-)
-
 type ErrEmptyCommand struct{}
 
 func (e *ErrEmptyCommand) Error() string {
 	return "Invalid spec for shell resource: command is empty."
 }
 
-// executionFailedError is used to help formatting errors reported to the user
-type executionFailedError struct {
-	ErrCode int
-	Stdout  string
-	Stderr  string
-	Command string
-}
-
-func (e *executionFailedError) Error() string {
-	return errorMessage(e.ErrCode, e.Command, e.Stdout, e.Stderr)
-}
+type ExecutionFailedError struct{}
```

Comment:
In fact, `error` is not a type but and inteface (https://tour.golang.org/methods/19). It means that you cannot instanciate an object of type `error`

Instead, you can call other methods such as `errors.New()`  which instanciate a new "basic error" which is a type verifying the `error` interface.

In fact, any custom type which implements a function `Error() string` verifies the interface `error` and can be used as such.

Due to the "logging" logic moving to the `shell.report()` method, this type is simplified in the PR to its strict minimum.
But it's kept because both `source` and `target` can throw this error.